packages/yourloops/components/patient-data.tsx

```diff
@@ -89,36 +92,50 @@ function PatientDataPage(): JSX.Element | null {
 
   const { blipApi } = dataHook
   const { patientId: paramPatientId = null } = paramHook as PatientDataParam
-  const authUser = authHook.user
-  const userId = authUser?.id ?? null
-  const userIsPatient = authHook.user?.isUserPatient()
-  const userIsHCP = authHook.user?.isUserHcp()
+  const userId = user?.id ?? null
+  const userIsPatient = user?.isUserPatient()
+  const userIsHCP = user?.isUserHcp()
   const prefixURL = userIsPatient ? '' : `/patient/${paramPatientId}`
 
-  const initialized = authHook.isLoggedIn && blipApi
+  const { selectedTeamId, selectTeam } = useSelectedTeamContext()
+  const { getMedicalTeams } = useTeam()
+  const isSelectedTeamMedical = user.isUserHcp() && getMedicalTeams().some((team: Team) => team.id === selectedTeamId)
+
+  useEffect(() => {
+    const patientTeams = getPatientById(paramPatientId)?.teams
+    const isPatientInSelectedTeam = patientTeams?.some((team: PatientTeam) => team.teamId === selectedTeamId)
+
+    if (!userIsHCP || isPatientInSelectedTeam) {
+      return
+    }
+    const defaultPatientTeamId = patientTeams[0].teamId
+    selectTeam(defaultPatientTeamId)
```

Comment:
After much thinking, I am starting to think that maybe the way of writing `useEffect()` should be different than the way of writing basic JS methods, because here I completely see your point... Just a thought for now, I'll give this topic some more thought on the next comment you'll post on my PRs 😄